internal/telemetry/client_test.go

```diff
@@ -140,6 +136,72 @@ func TestMetrics(t *testing.T) {
 	}
 }
 
+func TestDistributionMetrics(t *testing.T) {
+	t.Setenv("DD_TELEMETRY_HEARTBEAT_INTERVAL", "1")
+	var (
+		mu  sync.Mutex
+		got []DistributionSeries
+	)
+	closed := make(chan struct{}, 1)
+
+	// we will try to set one metric that the server must receive
+	expectedMetrics := 1
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		rType := RequestType(r.Header.Get("DD-Telemetry-Request-Type"))
+		if rType != RequestTypeDistributions {
+			return
+		}
+		req := Body{
+			Payload: new(DistributionMetrics),
+		}
+		dec := json.NewDecoder(r.Body)
+		err := dec.Decode(&req)
+		if err != nil {
+			t.Fatal(err)
+		}
+		v, ok := req.Payload.(*DistributionMetrics)
+		if !ok {
+			t.Fatal("payload set metrics but didn't get metrics")
+		}
+		mu.Lock()
+		defer mu.Unlock()
+		got = append(got, v.Series...)
+		if len(got) == expectedMetrics {
+			select {
+			case closed <- struct{}{}:
+			default:
+			}
+			return
+		}
+	}))
+	defer server.Close()
+
+	go func() {
+		client := &client{
+			URL: server.URL,
+		}
+		client.start(nil, NamespaceTracers)
+		// Records should have the most recent value
+		client.Record(NamespaceTracers, MetricKindDist, "soobar", 1, nil, false)
+		client.Record(NamespaceTracers, MetricKindDist, "soobar", 3, nil, false)
+		client.mu.Lock()
+		client.flush()
```

Comment:
Doesn't stop trigger a flush? So is this needed?

Reply:
I actually meant to remove `client.stop()` here, thanks for pointing this out!
I wanted to replace `stop` with a direct call to `flush` so this unit test doesn't rely on the functionality of `stop`. Implementation of `stop` may change in the future because of our conversation about only sending `app-closing` when `SIGINT` and `SIGQUIT` are intercepted.